lib/pbench/test/functional/server/test_user.py

```diff
@@ -67,25 +93,50 @@ def test_login(self, pbench: Pbench):
         does run tests within a single class in order. Is it worth taking on
         another dependency to make this explicit?
         """
-        pbench.login(self.USERNAME, self.PASSWORD)
+        pbench_server_client.login(USERNAME1, PASSWORD)
+        assert pbench_server_client.auth_token
 
-    def test_profile(self, pbench: Pbench):
+    def test_profile_bad_auth(self, pbench_server_client: PbenchServerClient):
+        """
+        Test that we can't access a user profile with an invalid authentication
+        token.
+        """
+        with pytest.raises(
+            HTTPError, match=r"UNAUTHORIZED for url: http://.*?/api/v1/user/functional"
+        ):
+            pbench_server_client.get(
+                "user",
+                {"target_username": USERNAME1},
+                headers={"Authorization": "Bearer tokens"},
```

Comment:
To expand...it would be good to have a test that non-lower-case keys work, but this test scenario is testing something else and shouldn't be exercising that.  (And, of course, if we were constructing a `CaseInsensitiveDict` here instead of a standard `Dict`, then it wouldn't matter whether the key was lowercase or not.)

Reply:
It's doing both rather than adding the bulk of additional test cases for minor unit test variations. 😄